cli/azd/internal/vsrpc/stream.go

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package vsrpc
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/gorilla/websocket"
+	"go.lsp.dev/jsonrpc2"
+)
+
+// wsStream adapts the websocket.Conn to jsonrpc2.Stream interface
+type wsStream struct {
+	c *websocket.Conn
+}
```

Comment:
Why do we need our own custom implementation of a websocket stream? Does the package not provide a default implementation or is there something we're doing that is custom?

Reply:
> Does the package not provide a default implementation or is there something we're doing that is custom?

The package does not provide one (or if it did, I strugged to find it).  JSON-RPC is transport agnostic and so the server package just handles the concerns above the transport level and requires something that implements their reader and writer interface.

We are not doing anything custom here - but the design of the Server API pushes us to create this.